src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,141 @@
+        User newUser = new User();
+        newUser.setLogin("someLogin1");
+        newUser.setPassword("short");
+        newUser.setAge(28);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(newUser));
+    }
+
+    @Test
+    void registerValidUser_Ok() {
+        User newUser = new User();
+        newUser.setLogin("someLogin2");
+        newUser.setPassword("Valid12345");
+        newUser.setAge(28);
+        User registeredUser = registrationService.register(newUser);
+        assertNotNull(registeredUser);
+        assertEquals(newUser.getLogin(), registeredUser.getLogin());
+        assertEquals(newUser.getPassword(), registeredUser.getPassword());
+        assertEquals(newUser.getAge(), registeredUser.getAge());
+    }
+
+    @Test
+    void registerAgeTooYoung_NotOk() {
+        User newUser = new User();
+        newUser.setLogin("YetAnotherUniqueLogin");
+        newUser.setPassword("ValidPassword123");
+        newUser.setAge(17);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(newUser));
+    }
+
+    @Test
+    void registerTooShortLogin_NotOk() {
+        User newUser = new User();
+        newUser.setLogin("Pasha");
+        newUser.setPassword("ValidPassword");
+        newUser.setAge(28);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(newUser));
+    }
+
+    @Test
+    void registerNullUser_NotOk() {
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(null));
+    }
+
+    @Test
+    void registerLoginContainWhiteSpace_NotOk() {
+        User newUser = new User();
+        newUser.setLogin("Pasha Otroda");
+        newUser.setPassword("ValidPassword");
+        newUser.setAge(28);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(newUser));
+    }
+
+    @Test
+    void registerNullLogin_NotOk() {
+        User newUser = new User();
+        newUser.setLogin(null);
+        newUser.setPassword("ValidPassword");
+        newUser.setAge(28);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(newUser));
+    }
+
+    @Test
+    void registerNullPassword_NotOk() {
+        User newUser = new User();
+        newUser.setLogin("someLogin1");
+        newUser.setPassword(null);
+        newUser.setAge(28);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(newUser));
+    }
+
+    @Test
+    void registerNullAge_NotOk() {
+        User newUser = new User();
+        newUser.setLogin("PashaOtroda");
+        newUser.setPassword("ValidPassword");
+        newUser.setAge(null);
+        assertThrows(RegistrationException.class, () ->
+                registrationService.register(newUser));
+    }
+
+    @Test
+    void registerNegativeAge_NotOk() {
```

Comment:
```suggestion
    void register_negativeAge_notOk() {
```
better adhere to this suggested naming convention